packages/sui-studio/src/runtime-mocha/index.js

```diff
@@ -0,0 +1,114 @@
+
+import SUIContext from '@s-ui/react-context'
+import withContext from '../components/demo/HoC/withContext'
+import {cleanDisplayName} from '../components/demo/utilities'
+import {requireFile, tryRequireContext, tryRequireComponent} from '../components/tryRequire'
+
+const originalKarmaLoader = window.__karma__.loaded
+window.__karma__.loaded = () => {}
+
+const regex = /\.\/(?<categoryName>\w+)\/(?<componentName>\w+)\/(src)+\/(index.js|jsx)$/
+
+function declareAll(resolve) {
+  const cache = {}
+  const capitalize = s => {
+    if (typeof s !== 'string') return ''
+    return s.charAt(0).toUpperCase() + s.slice(1)
+  }
+  resolve.keys().forEach(srcKey => {
+    const {
+      groups: {categoryName, componentName}
+    } = srcKey.match(regex)
+    const component = resolve(srcKey).default
+    const displayName = capitalize(categoryName) + capitalize(componentName)
+    cache[displayName] = {
+      displayName,
+      categoryName,
+      componentName
+    }
+  })
+  return cache
+}
+function testAll(resolve) {
+  resolve.keys().forEach(key => {
+    resolve(key)
+  })
+}
+
+const enhanceComponent = displayName => {
+  !displayName && console.error('[sui-Test] Component without displayName') // eslint-disable-line
+
+  const activeContext = window.__STUDIO_CONTEXTS__[displayName]
+  const ActiveComponent = window.__STUDIO_COMPONENT__[displayName]
+
+  const EnhanceComponent = withContext(
+    activeContext.default,
+    activeContext
+  )(ActiveComponent)
+
+  const NextComponent = props => (
+    <SUIContext.Provider value={activeContext.default}>
+      <EnhanceComponent {...props} />
+    </SUIContext.Provider>
+  )
+  hoistNonReactStatics(NextComponent, ActiveComponent)
+
+  const nextDisplayName = cleanDisplayName(displayName)
+  NextComponent.displayName = nextDisplayName
+
+  return NextComponent
+}
+
+const run = async () => {
+  const cache = declareAll(
+    require.context(
+      `${__BASE_DIR__}/components/`,
+      true,
+      /\.\/(\w+)\/(\w+)\/(src)+\/index.(js|jsx)$/
+    )
+  )
+
+  async function asyncForEach(array, callback) {
+    const response = []
+    for (let index = 0; index < array.length; index++) {
+      response[index] = await callback(array[index], index, array)
+    }
+    return response
+  }
+
+  window.__STUDIO_CONTEXTS__ = {}
+  window.__STUDIO_COMPONENT__ = {}
+  await asyncForEach(Object.keys(cache), async cachedValueKeyName => {
+    const {categoryName, componentName, displayName} = cache[cachedValueKeyName]
+    let nextEntityContexts = await tryRequireContext({
+      category: categoryName,
+      component: componentName
+    })
+    nextEntityContexts =
+      typeof nextEntityContexts !== 'function'
+        ? nextEntityContexts
+        : await nextEntityContexts() // eslint-disable-line
+    const nextEntityComponent = await tryRequireComponent({
+      category: categoryName,
+      component: componentName
+    })
+    window.__STUDIO_CONTEXTS__[displayName] = nextEntityContexts
```

Comment:
While development, sometimes we wrap a component in `React.memo` which removes `displayName`.
Is it possible to make a warning if displayName does not exist?

Reply:
we realized that and fixed it using
`const displayName = capitalize(categoryName) + capitalize(componentName)`
line 26

Reply:
the display name is provided by the babelPatcher AST plugin